src/sentry/sentry_apps/token_exchange/refresher.py

```diff
@@ -106,7 +119,7 @@ def application(self) -> ApiApplication:
         try:
             return ApiApplication.objects.get(client_id=self.client_id)
         except ApiApplication.DoesNotExist:
-            raise SentryAppIntegratorError(
+            raise SentryAppSentryError(
```

Comment:
Making this consistent with `_validator`, this step could fail because either the user entered the wrong client id or our db actually borked. I leaned to the latter to be a bit more defensive but we can see later how common this issue is.